src/common/helpers/glide.ts

```diff
@@ -0,0 +1,35 @@
+import { BUNDLER_ADDRESS, ID_REGISTRY_ADDRESS, bundlerABI } from "@farcaster/hub-web";
+import { createGlideClient } from "@paywithglide/glide-js";
+import { mainnet, base, optimism, polygon, arbitrum } from "@wagmi/core/chains";
+import { encodeFunctionData, toHex } from "viem";
+import { useReadContract } from "wagmi";
+import { PaymentOption } from "node_modules/@paywithglide/glide-js/dist/types";
+import { WARPCAST_RECOVERY_PROXY, getDeadline, readNoncesFromKeyGateway } from "./farcaster";
+
+export type PaymentOptions = PaymentOption[];
+
+export const glideClient = createGlideClient({
+  projectId: process.env.NEXT_PUBLIC_GLIDE_PROJECT_ID || "",
+  chains: [optimism, mainnet, base, arbitrum, polygon],
+});
+
+export const getGlidePaymentOptions = async (address: `0x${string}`, price: bigint): Promise<PaymentOptions> => {
+  if (price) {
+    return await glideClient.listPaymentOptions({
+      transaction: {
+        chainId: "eip155:10",
+        to: BUNDLER_ADDRESS,
+        value: toHex(0n),
+        input: encodeFunctionData({
+          abi: bundlerABI,
+          functionName: "price",
```

Comment:
this would require them to select payment _after_ they sign the transaction (to get the signature for the 'sig' param, or to sign twice (once to get the payment options and once to pay), though, is this intentional?